float32    x
float32    y
bool       is_heading